kyo-core/shared/src/main/scala/kyo/logs.scala

```diff
@@ -13,6 +15,11 @@ object Logs extends logsPlatformSpecific:
         def warnEnabled: Boolean
         def errorEnabled: Boolean
 
+        inline def trace(inline params: Param[?]*)(
+            using inline file: internal.Position
+        ): Unit =
+            trace(Param.show(params*))
```

Comment:
I don't see the value of this. When would I ever want to print code explicitly?

It may be useful to have an arbitrary `.trace` function that could be invoked on any Kyo effect...However, that would involve an extension method.

Reply:
It's a convenience to make it easy to log labeled values. Example:

```scala
// Before
def test(person: Person, address: Address) =
     Logs.debug(s"person=$person, address=$address")

// After
def test(person: Person, address: Address) =
     Logs.debug(person, address) // Outputs `Params(person -> $person, address -> $address)`
```

Reply:
Can't that be done with accepting `Any` and a `.toString`? Why do we need a macro?

Reply:
I don't think you'd be able to produce the labels?

Reply:
To exemplify, I've been using this to aid in debugging the new `Tag` implementation. It's quite convenient to add log messages and produce meaningful insights into the execution:

```scala
def checkType(
    subTag: Tag[?],
    superTag: Tag[?],
    subIdx: Int,
    superIdx: Int,
    outerVariance: Variance,
    innerVariance: Variance
): Boolean =
    // I could have used a string interpolation here to show what each printed value is
    // but with this new mechanism, I just need to pass the objects I want to inspect
    // and the log will automatically show labels for each value.
    Logs.unsafe.debug(subTag.className, superTag.className, outerVariance, innerVariance)
    val variance =
        (outerVariance, innerVariance) match
            case (Invariant, Invariant)         => Invariant
     ....
```

This produces this output:

```
[info]   - T1 <:< T2 (2 milliseconds)
19:24:48.915 [scala-execution-context-global-25] DEBUG kyo.logs -- [tags.scala:228 - method checkType] Params(
  "subTag.className" -> "kyoTest.TagsTest._$Test",
  "superTag.className" -> "kyoTest.TagsTest._$Test",
  "outerVariance" -> Covariant,
  "innerVariance" -> Covariant
)
19:24:48.916 [scala-execution-context-global-25] DEBUG kyo.logs -- [tags.scala:228 - method checkType] Params(
  "subTag.className" -> "kyoTest.TagsTest._$Test",
  "superTag.className" -> "kyoTest.TagsTest._$Test",
  "outerVariance" -> Covariant,
  "innerVariance" -> Contravariant
)
19:24:48.916 [scala-execution-context-global-25] DEBUG kyo.logs -- [tags.scala:228 - method checkType] Params(
  "subTag.className" -> "kyoTest.TagsTest._$Test",
  "superTag.className" -> "kyoTest.TagsTest._$Test",
  "outerVariance" -> Covariant,
  "innerVariance" -> Covariant
)
19:24:48.916 [scala-execution-context-global-25] DEBUG kyo.logs -- [tags.scala:228 - method checkType] Params(
  "subTag.className" -> "kyoTest.TagsTest._$Test",
  "superTag.className" -> "kyoTest.TagsTest._$Test",
  "outerVariance" -> Covariant,
  "innerVariance" -> Contravariant
)
19:24:48.916 [scala-execution-context-global-25] DEBUG kyo.logs -- [tags.scala:228 - method checkType] Params(
  "subTag.className" -> "kyoTest.TagsTest._$Test",
  "superTag.className" -> "kyoTest.TagsTest._$Test",
  "outerVariance" -> Covariant,
  "innerVariance" -> Covariant
)
19:24:48.916 [scala-execution-context-global-25] DEBUG kyo.logs -- [tags.scala:228 - method checkType] Params(
  "subTag.className" -> "kyoTest.TagsTest._$Test",
  "superTag.className" -> "kyoTest.TagsTest._$Test",
  "outerVariance" -> Covariant,
  "innerVariance" -> Covariant
)
[info]   - T2 <:< T1 *** FAILED *** (1 millisecond)
[info]     false did not equal true Tag[T2] <:< Tag[T1] is false but ITag[T2] <:< ITag[T1] is true (tagsTest.scala:18)
```